packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,240 @@
+<line_data_without_label>"]"" "*          {this.popState(); this.popState()}
+<line>[^\s\[]+" "*                          {this.begin("line_data"); return 'LINE_TITLE';}
+<line>"["" "*                             {this.begin('line_data_without_label');}
+
+"bar"" "*                                 {this.begin("bar"); return 'BAR';}
+<bar>["]                                  {this.begin("bar_title");}
+<bar_title>[^"]+                          {return 'BAR_TITLE';}
+<bar_title>["]" "*                        {this.popState(); this.begin("bar_data");}
+<bar_data>"["" "*                         {this.begin('bar_data_entries');}
+<bar_data_without_label,bar_data_entries>(?:[+-]?\d+(?:\.\d+)?)+(?:" "*[,]" "*(?:[+-]?\d+(?:\.\d+)?)+)*" "*    {return 'BAR_DATA'}
+<bar_data_entries>"]"" "*                 {this.popState(); this.popState(); this.popState()}
+<bar_data_without_label>"]"" "*           {this.popState(); this.popState()}
+<bar>[^\s\[]+" "*                           {this.begin("bar_data"); return 'BAR_TITLE';}
+<bar>"["" "*                              {this.begin('bar_data_without_label');}
+
+
+
+
+["][`]                                    { this.begin("md_string");}
+<md_string>[^`"]+                         { return "MD_STR";}
+<md_string>[`]["]                         { this.popState();}
+["]                                       this.begin("string");
+<string>["]                               this.popState();
+<string>[^"]*                             return "STR";
+
+[A-Za-z]+                                 return 'ALPHA';
+":"                                       return 'COLON';
+\+                                        return 'PLUS';
+","                                       return 'COMMA';
+"="                                       return 'EQUALS';
+\=                                        return 'EQUALS';
+"*"                                       return 'MULT';
+\#                                        return 'BRKT';
+[\_]                                      return 'UNDERSCORE';
+"."                                       return 'DOT';
+"&"                                       return 'AMP';
+\-                                        return 'MINUS';
+[0-9]+                                    return 'NUM';
+\s                                        return 'SPACE';
+";"                                       return 'SEMI';
+[!"#$%&'*+,-.`?\\_/]                      return 'PUNCTUATION';
+<<EOF>>                                   return 'EOF';
+
+/lex
+
+%start start
+
+%% /* language grammar */
+
+start
+  : eol start
+  | SPACE start
+  | directive start
+  | XYCHART chartConfig CHART_CONFIG_END document
+	| XYCHART CHART_CONFIG_END document
+	;
+
+chartConfig
+  : chart_orientation {yy.setOrientation($1.trim());}
+  ;
+
+document
+	: /* empty */
+	| document line
+	;
+
+line
+	: statement eol
+	;
+
+statement
+  :
+  | SPACE statement
```

Comment:
Ignoring spaces would make parsing much easier. You needed this
```
statement: SPACE statement
```
to mitigate that problem.